plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/WorkflowCpsHooks.java

```diff
@@ -0,0 +1,64 @@
+package org.jenkins.tools.test.hook;
+
+import hudson.model.UpdateSite;
+import hudson.util.VersionNumber;
+import java.util.List;
+import java.util.Map;
+import org.jenkins.tools.test.model.PomData;
+import org.jenkins.tools.test.model.hook.PluginCompatTesterHookBeforeExecution;
+
+public class WorkflowCpsHooks {
+    public static boolean isMultiModuleVersionOfWorkflowCps(Map<String, Object> info) {
+        PomData data = (PomData) info.get("pomData");
+        UpdateSite.Plugin plugin = info.get("plugin") != null ? (UpdateSite.Plugin) info.get("plugin") : null;
+        return isMultiModuleVersionOfWorkflowCps(plugin);
+    }
+
+    public static boolean isMultiModuleVersionOfWorkflowCps(UpdateSite.Plugin plugin) {
+        if (plugin != null && plugin.name.equalsIgnoreCase("workflow-cps") && plugin.version != null) {
+            VersionNumber pluginVersion = new VersionNumber(plugin.version);
+            VersionNumber multiModuleSince = new VersionNumber("2803.v1a_f77ffcc773");
+            return pluginVersion.isNewerThan(multiModuleSince);
+        }
+        return false;
+    }
+
+    public static class MultiModuleHook extends AbstractMultiParentHook {
+        @Override
+        protected String getParentFolder() {
+            return "workflow-cps-plugin";
+        }
+
+        @Override
+        protected String getParentProjectName() {
+            return "workflow-cps-parent";
+        }
+
+        @Override
+        protected String getPluginFolderName(UpdateSite.Plugin currentPlugin){
+            return "plugin";
+        }
+
+        @Override
+        public boolean check(Map<String, Object> info) {
+            return isMultiModuleVersionOfWorkflowCps(info);
+        }
+    }
+
+    /**
+     * Do not run tests for groovy-cps-dgm-builder and groovy-cps in the PCT.
+     */
+    public static class OnlyTestPluginHook extends PluginCompatTesterHookBeforeExecution {
+        @Override
+        public Map<String, Object> action(Map<String, Object> info) throws Exception {
+            if (isMultiModuleVersionOfWorkflowCps(info)) {
+                List<String> args = (List<String>) info.get("args");
+                if (args != null) {
+                    args.add("-pl");
+                    args.add("plugin");
```

Comment:
Running groovy-cps's tests is quick and relatively harmless, but since it doesn't have plugin dependencies, testing it in the PCT is not very useful. The main reason for this is that without it, testing `groovy-cps` fails because `groovy-cps-dgm-builder` cannot be found in the local Maven repository due to https://github.com/jenkinsci/workflow-cps-plugin/pull/612/commits/1b01482f7077021d68c23b1c266e3b669ff83182. Another option would be be to modify the goal to `install` and recompile the plugin and the libraries a second time so that we can test everything.

Reply:
> modify the goal to `install`

I thought that was what you were already doing in this PR. :shrug: whatever works

Reply:
The PCT compiles the plugin with one invocation of Maven and then runs tests separately with another. Compilation typically runs `process-test-classes`, but for multi-module projects (and `workflow-cps` as of the new logic in this PR) runs `install`. Tests are normally run using individual test goals like `surefire:test` (and this PR tweaks that invocation to add `-pl plugin` for `workflow-cps`). I am not sure why compilation and testing is split like this, but I do not work with the PCT very much.

Reply:
> I am not sure why compilation and testing is split like this

It is mostly a relic of the old mode of PCT predating https://github.com/jenkinsci/maven-hpi-plugin/pull/336. The idea was to be able to `mvn test` or `mvn install` in one pass with suitable properties. Unfortunately for technical reasons you need to define `jenkins.version` during at least the test phase, meaning that the _compilation_ would also be against the new version, which we do not want. So we would want a PCT run (incl. multimodule) to look like

```bash
mvn -Pquick-build install
mvn -Djenkins.version=… -Dwhatever.war.arg=… hpi:resolve-test-dependencies surefire:test
```

(so that the second command does not recompile anything).